### PR TITLE
linux-x86-clang target: Add -latomic

### LINKED ISSUE #16572: openssl-3.0.0: -latomic missing from linux-x86-clang
Trying to build openssl-3.0.0 with `./Configure linux-x86-clang && make` fails with undefined references:
```
${LDCMD:-clang} -pthread -m32 -Wa,--noexecstack -Qunused-arguments -Wall -O3 -fomit-frame-pointer -L.   \
        -o apps/openssl \
        apps/lib/openssl-bin-cmp_mock_srv.o \
        apps/openssl-bin-asn1parse.o apps/openssl-bin-ca.o \
        apps/openssl-bin-ciphers.o apps/openssl-bin-cmp.o \
        apps/openssl-bin-cms.o apps/openssl-bin-crl.o \
        apps/openssl-bin-crl2pkcs7.o apps/openssl-bin-dgst.o \
        apps/openssl-bin-dhparam.o apps/openssl-bin-dsa.o \
        apps/openssl-bin-dsaparam.o apps/openssl-bin-ec.o \
        apps/openssl-bin-ecparam.o apps/openssl-bin-enc.o \
        apps/openssl-bin-engine.o apps/openssl-bin-errstr.o \
        apps/openssl-bin-fipsinstall.o apps/openssl-bin-gendsa.o \
        apps/openssl-bin-genpkey.o apps/openssl-bin-genrsa.o \
        apps/openssl-bin-info.o apps/openssl-bin-kdf.o \
        apps/openssl-bin-list.o apps/openssl-bin-mac.o \
        apps/openssl-bin-nseq.o apps/openssl-bin-ocsp.o \
        apps/openssl-bin-openssl.o apps/openssl-bin-passwd.o \
        apps/openssl-bin-pkcs12.o apps/openssl-bin-pkcs7.o \
        apps/openssl-bin-pkcs8.o apps/openssl-bin-pkey.o \
        apps/openssl-bin-pkeyparam.o apps/openssl-bin-pkeyutl.o \
        apps/openssl-bin-prime.o apps/openssl-bin-progs.o \
        apps/openssl-bin-rand.o apps/openssl-bin-rehash.o \
        apps/openssl-bin-req.o apps/openssl-bin-rsa.o \
        apps/openssl-bin-rsautl.o apps/openssl-bin-s_client.o \
        apps/openssl-bin-s_server.o apps/openssl-bin-s_time.o \
        apps/openssl-bin-sess_id.o apps/openssl-bin-smime.o \
        apps/openssl-bin-speed.o apps/openssl-bin-spkac.o \
        apps/openssl-bin-srp.o apps/openssl-bin-storeutl.o \
        apps/openssl-bin-ts.o apps/openssl-bin-verify.o \
        apps/openssl-bin-version.o apps/openssl-bin-x509.o \
        apps/libapps.a -lssl -lcrypto -ldl -pthread 
/usr/bin/x86_64-pc-linux-gnu-ld: ./libcrypto.so: undefined reference to `__atomic_is_lock_free'
/usr/bin/x86_64-pc-linux-gnu-ld: ./libcrypto.so: undefined reference to `__atomic_fetch_or_8'
/usr/bin/x86_64-pc-linux-gnu-ld: ./libcrypto.so: undefined reference to `__atomic_load'
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
```

Interestingly, using `./Configure linux-x86`, which uses gcc instead of clang works correctly. So does `./Configure linux-x86-clang -latomic`, it looks like clang needs libatomic while gcc doesn't. I see that there were changes with the handling of -latomic, #15086, but partially reverted in #15640, the latter suggesting simply documenting the incomplete handling of this library, so I'm not sure what would be the correct action to fix this.

OS: Gentoo Linux
Clang: 12.0.0
Gcc: 9.3.0
[perl configdata.pm --dump](https://github.com/openssl/openssl/files/7137865/dump.txt)

### PR DESCRIPTION
**Pull request by t8m**
Fixes #16572

### CODE REVIEW

**Review by paulidale — APPROVED** [review]
LGTM for master and 3.0

File: .github/workflows/compiler-zoo.yml [hunk 1]
```diff
@@ -38,11 +30,12 @@ jobs:
       run: |
         sudo apt-get update
         sudo apt-get -yq --force-yes install ${{ matrix.zoo.cc }}
+        sudo apt-get -yq --force-yes install gcc-multilib
```
Comment (levitte) [inline .github/workflows/compiler-zoo.yml:33]: Uhm, how are the different clang versions installed?

Comment (t8m) [reply .github/workflows/compiler-zoo.yml:33]: By the command above?

Comment (paulidale) [reply .github/workflows/compiler-zoo.yml:33]: `apt-get install clang-12` and similar.  They are later invoked including the version so it should all be happy.

**Comment by t8m** [comment]
Merged to master and 3.0 branches. Thank you for the review.

**Pull request closed without merging**